src/main/java/com/coreos/jetcd/data/ByteSequence.java

```diff
@@ -0,0 +1,121 @@
+package com.coreos.jetcd.data;
+
+import com.google.protobuf.ByteString;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.CharBuffer;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+
+/**
+ * Etcd binary bytes, easy to convert between byte[], String and ByteString.
+ */
+public class ByteSequence {
+
+    private final byte[] bytes;
+    private final int hashVal;
+    private final ByteString byteString;
```

Comment:
I prefer to remove bytes[] here.

Currently, both of byte[] and ByteString are hold in ByteSequence.  I think there are no need to hold byte[] since all method only use ByteString except ```calcHashCore```.  However, ```calcHashCore``` can use ```source``` and I think it's enough. What do you think?